atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBackupRunner.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import org.apache.cassandra.thrift.Cassandra;
+import org.apache.cassandra.thrift.Compression;
+import org.apache.cassandra.thrift.ConsistencyLevel;
+import org.apache.cassandra.thrift.CqlResult;
+import org.apache.thrift.TException;
+import org.immutables.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.common.annotation.Idempotent;
+import com.palantir.common.base.Throwables;
+import com.palantir.util.Pair;
+
+public class CassandraTimestampBackupRunner {
+    private static final Logger log = LoggerFactory.getLogger(CassandraTimestampBackupRunner.class);
+
+    private final CassandraKeyValueService cassandraKeyValueService;
+
+    public CassandraTimestampBackupRunner(CassandraKeyValueService cassandraKeyValueService) {
+        this.cassandraKeyValueService = cassandraKeyValueService;
+    }
+
+    /**
+     * Creates the timestamp table, if it doesn't already exist.
+     */
+    @Idempotent
+    public void createTimestampTable() {
+        cassandraKeyValueService.createTable(
+                AtlasDbConstants.TIMESTAMP_TABLE,
+                CassandraTimestampUtils.TIMESTAMP_TABLE_METADATA.persistToBytes());
+    }
+
+    /**
+     * Writes a backup of the existing timestamp to the database, if none exists. After this backup, this timestamp
+     * service can no longer be used until a restore. Note that the value returned is the value that was backed up;
+     * multiple calls to this method are safe, and will return the backed up value.
+     *
+     * This method assumes that the timestamp table exists.
```

Comment:
I've put this at the top of `getCurrentBoundData()` instead which is called by both.

(Surprisingly this was not entirely trivial since most of the getTables() methods exposed by CassandraKVS filter out hidden tables!)